src/circuit/value_commit_orchard.rs

```diff
@@ -0,0 +1,391 @@
+                        magnitude_zs[magnitude_words].clone(),
+                        magnitude_extra_bits, // The 7th window must be a 4 bits value.
+                    )?;
+                }
+
+                // Multiply asset by magnitude, using the long scalar mul.
+                // TODO: implement a new variable base multiplication which is optimized for 64-bit scalar
+                // (the long scalar mul is optimized for pallas::Base scalar (~255-bits))
+                //
+                // magnitude_asset = [magnitude] asset
+                let magnitude_asset = {
+                    let magnitude_scalar = ScalarVar::from_base(
+                        ecc_chip.clone(),
+                        layouter.namespace(|| "magnitude"),
+                        &v_net_magnitude_sign.0,
+                    )?;
+                    let (magnitude_asset, _) = params
+                        .asset
+                        .mul(layouter.namespace(|| "[magnitude] asset"), magnitude_scalar)?;
+                    magnitude_asset
+                };
+
+                // commitment = [sign] magnitude_asset = [v_net_magnitude_sign] asset
+                magnitude_asset.mul_sign(
+                    layouter.namespace(|| "[sign] commitment"),
+                    &v_net_magnitude_sign.1,
+                )?
+            }
+        };
+
+        // blind = [rcv] ValueCommitR
+        let (blind, _rcv) = {
+            let value_commit_r = OrchardFixedBasesFull::ValueCommitR;
+            let value_commit_r = FixedPoint::from_inner(ecc_chip, value_commit_r);
+
+            // [rcv] ValueCommitR
+            value_commit_r.mul(layouter.namespace(|| "[rcv] ValueCommitR"), rcv)?
+        };
+
+        // [v] ValueCommitV + [rcv] ValueCommitR
+        commitment.add(layouter.namespace(|| "cv"), &blind)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        circuit::{
+            gadget::assign_free_advice,
+            value_commit_orchard::{gadgets::value_commit_orchard, ZsaValueCommitParams},
+            K,
+        },
+        constants::{OrchardCommitDomains, OrchardFixedBases, OrchardHashDomains},
+        note::AssetBase,
+        value::{NoteValue, ValueCommitTrapdoor, ValueCommitment},
+    };
+    use halo2_gadgets::{
+        ecc::{
+            chip::{EccChip, EccConfig},
+            NonIdentityPoint, ScalarFixed,
+        },
+        sinsemilla::chip::{SinsemillaChip, SinsemillaConfig},
+        utilities::lookup_range_check::{
+            LookupRangeCheck4_5BConfig, PallasLookupRangeCheck4_5BConfig,
+        },
+    };
+
+    use group::Curve;
+    use halo2_proofs::{
+        circuit::{Layouter, SimpleFloorPlanner, Value},
+        dev::MockProver,
+        plonk::{Advice, Circuit, Column, ConstraintSystem, Error, Instance},
+    };
+    use pasta_curves::pallas;
+
+    use rand::{rngs::OsRng, RngCore};
+
+    #[test]
+    fn test_value_commit_orchard() {
```

Comment:
```suggestion
    fn test_value_commit_orchard_zsa() {
```
because the test hard-codes `Some(ZsaValueCommitParams { .. })` and thus is only testing the OrchardZSA version of `value_commit_orchard()`.